docs/src/basics/structs_tuples_and_enums.md

```diff
@@ -103,34 +120,56 @@ enum InventoryEvent {
 }
 ```
 
+If we wanted to instantiate an enum with some interior data, it looks like this:
+
 ```sway
-enum Color {
-    Blue: (),
-    Green: (),
-    Red: (),
-    Silver: (),
-    Grey: (),
+fn main() {
+  let event = InventoryEvent::ItemLoss(Claim {
+      insurance_company: ~Insurer::default(),
+      item_number: 42,
+      item_cost: 1_000,
+  });
 }
+```
 
-fn main() {
-    let color = Color::Blue;
+### Enums of Enums
+
+It is possible to define enums of enums for example
+
+```sway
+enum Error {
+    StateError: StateError,
+    UserError: UserError,
+}
+
+enum StateError {
+    Void: (),
+    Pending: (),
+    Completed: (),
+}
+
+enum UserError {
+    InsufficientPermissions: (),
+    Unauthorized: (),
 }
 ```
 
-Here, we have instantiated a variable named `color` with _enum instantiation syntax_. Note that enum instantiation does not require the `~` tilde syntax. If we wanted to instantiate an enum with some interior data, it looks like this:
+To use them the syntax is
 
 ```sway
-struct Claim {
-    insurance_company: Insurer,
-    item_number: u64,
-    item_cost: u64,
+fn main() {
+    let error1 = Error::StateError(StateError::Void);
+    let error2 = Error::UserError(UserError::Unauthorized);
 }
+```
 
-let event = InventoryEvent::ItemLoss(Claim {
-    insurance_company: ~Insurer::default(),
-    item_number: 42,
-    item_cost: 1_000,
-});
+The syntax above is not the most ergonomic but it is possible. It is advisable to not use the `enum Error` structure which nests `enums` and instead use the `enums` directly
```

Comment:
What's the use of `Error` then if it's never used?

Reply:
Initially I thought that I could use the syntax `Error::StateError::Void`.
I was thinking about a hierarchy of enums and "sub-classing" them in a way which made me get to that nasty syntax.
Since the issue was around I thought to mention it plus since enums containing structs is a thing then someone might wonder what enum of enums is like.
That being said, it's nasty so I'm pointing out that it's better to not use it and instead have simple enums.

Reply:
Ah I see. Then maybe clarify that a bit more, that it's easier to not use deeply-nested enums.